Blog: Announce maintainer projects (#892)
Homebrew/brew.sh: _posts/2022-09-15-homebrew-maintainer-projects-update.md

---
title: Maintainer Projects
author: SMillerDev
---

> Homebrew’s Project Leadership Committee has green-lit two paid projects by our maintainers this year and since both have hit some milestones recently we’d love to give you, our sponsors and users, an update on the progress.

Reviewer: ```suggestion
Homebrew’s Project Leadership Committee has green-lit two paid projects by our maintainers this year and since both have hit some milestones recently we’d love to give you, our sponsors and users, an update on their progress.
```

> Homebrew’s Project Leadership Committee has green-lit two paid projects by our maintainers this year and since both have hit some milestones recently we’d love to give you, our sponsors and users, an update on the progress.

Reviewer: ```suggestion
Homebrew’s Project Leadership Committee has green-lit two paid projects by our maintainers this year and since both have hit some milestones recently we’d love to give you, our sponsors and users, an update on their progress.
```

## API Project

> Since version [3.6.0](./2022-09-07-homebrew-3.6.0.md) of Homebrew, we are now choosing to advertise the `HOMEBREW_INSTALL_FROM_API` variable to the general public. A lot of work was done by @rylan to make the experience more stable and the API install method is bound to bring a big performance improvement to those who dare enable it.

Author: ```suggestion
Since version [3.6.0](./2022-09-07-homebrew-3.6.0.md) of Homebrew, we are now choosing to advertise the `HOMEBREW_INSTALL_FROM_API` variable to the general public. A lot of work was done by @Rylan12 to make the experience more stable and the API install method is bound to bring a big performance improvement to those who dare enable it.
```

> Since version [3.6.0](./2022-09-07-homebrew-3.6.0.md) of Homebrew, we are now choosing to advertise the `HOMEBREW_INSTALL_FROM_API` variable to the general public. A lot of work was done by @Rylan12 to make the experience more stable and the API install method is bound to bring a big performance improvement to those who dare enable it.

Reviewer 2: ```suggestion
Since version [3.6.0](./2022-09-07-homebrew-3.6.0.md) of Homebrew, we are now choosing to advertise the `HOMEBREW_INSTALL_FROM_API` environment variable to the general public. A lot of work was done by @Rylan12 to make the experience more stable and the API install method is bound to bring a big performance improvement to those who dare enable it.
```

> The goal of the HOMEBREW_INSTALL_FROM_API environment variable is to allow Homebrew users to install formulae and casks without needing to have Homebrew/homebrew-core and/or Homebrew/homebrew-cask tapped. This is intended to save space on the user’s machine and also to make commands like brew update much faster.

Reviewer 2: ```suggestion
The goal of the `HOMEBREW_INSTALL_FROM_API` environment variable is to allow Homebrew users to install formulae and casks without needing to have Homebrew/homebrew-core and/or Homebrew/homebrew-cask tapped. This is intended to save space on the user’s machine and also to make commands like `brew update` much faster.
```

Reviewer 2: Not sure if the tap names should also be in backticks

> The goal of the HOMEBREW_INSTALL_FROM_API environment variable is to allow Homebrew users to install formulae and casks without needing to have Homebrew/homebrew-core and/or Homebrew/homebrew-cask tapped. This is intended to save space on the user’s machine and also to make commands like brew update much faster.

Reviewer 3: ```suggestion
The goal of the `HOMEBREW_INSTALL_FROM_API` environment variable is to allow Homebrew users to install formulae and casks without needing to have Homebrew/homebrew-core and/or Homebrew/homebrew-cask tapped. This is intended to make `brew update` much faster and required less often and also to save space on the user’s machine .
```

> Now that the variable is public, we’re going into a period of bug hunting. When the users stop finding problems and edge-cases for the API to handle, we will be making this the default for all users so everyone can enjoy a fast Homebrew experience.

Reviewer: ```suggestion
Now that the variable is public, we’re going into a period of bug hunting. When its users stop finding problems and edge-cases for the API to handle, we will be making this the default for all users so everyone can enjoy a fast Homebrew experience.
```

## Ephemeral CI Runners

> Since July 18th the Intel based CI runners are now ephemeral and the basis has been laid for the same features for Apple Silicon based runners. While Homebrew users might not notice more than sleeping a little easier, I can tell you that the maintainers are very happy with this improvement and the accompanying dashboard.

Reviewer: ```suggestion
Since July 18th the Intel-based CI runners are now ephemeral and the foundation has been laid for applying the same features to Apple Silicon based runners. While Homebrew users might not notice more than sleeping a little easier thanks to bottling runs being more reliable, I can tell you that the maintainers are very happy with this improvement and the accompanying dashboard.
```
(I doubt the average Homebrew user would actually notice at all, but I've added a potential reason)

Author: Yeah, I was going for them sleeping easier because of the better security.

> Currently our runners persist indefinitely, which has some distinct downsides:

Reviewer: ```suggestion
Currently our runners persist indefinitely, which has some distinct downsides:

```

- Security is worse since a compromised runner will keep working indefinitely

> - Keeping them maintained and up to date is harder

Reviewer 4: ```suggestion
- Keeping them maintained is harder
```

The update process hasn't changed much. But I agree with the general notion of not having to deal with stuck runners etc anymore.

- We cannot dynamically scale up/down usage per-OS

> The goal was to replace all our persistent runners with ephemeral variants, starting with the Apple Silicon runners. Unfortunately we ran into some bugs in the Okta platform Macstadium provides for us, so that initial goal had to be shifted to Intel runners first.

Reviewer 5: ```suggestion
The goal was to replace all our persistent runners with ephemeral variants, starting with the Apple Silicon runners. Unfortunately we ran into some bugs in the Orka platform MacStadium provides for us, so that initial goal had to be shifted to Intel runners first.
```

> The goal was to replace all our persistent runners with ephemeral variants, starting with the Apple Silicon runners. Unfortunately we ran into some bugs in the Okta platform Macstadium provides for us, so that initial goal had to be shifted to Intel runners first.

Reviewer 6: ```suggestion
The goal was to replace all our persistent runners with ephemeral variants, starting with the Apple Silicon runners. Unfortunately we ran into some bugs in the Orka platform MacStadium provides for us, so that initial goal had to be shifted to Intel runners first.
```

> The goal was to replace all our persistent runners with ephemeral variants, starting with the Apple Silicon runners. Unfortunately we ran into some bugs in the Orka platform MacStadium provides for us, so that initial goal had to be shifted to Intel runners first.

Reviewer: ```suggestion
The goal was to replace all our persistent runners with ephemeral variants, starting with the Apple Silicon runners. Unfortunately we ran into some bugs in the Orka platform which MacStadium provides for us, so that initial goal had to be shifted to Intel runners first.
```

> In the future we hope to enable the Apple Silicon based runners as ephemeral runners to close this project from the macOS side. Once that is done the stretch goal for this project is to make our Linux self hosted runners ephemeral too.

Reviewer 6: ```suggestion
In the future we hope to enable the Apple Silicon based runners as ephemeral runners to close this project from the macOS side. Once that is done the stretch goal for this project is to make our self-hosted Linux runners ephemeral too.
```

> In the future we hope to enable the Apple Silicon based runners as ephemeral runners to close this project from the macOS side. Once that is done the stretch goal for this project is to make our self-hosted Linux runners ephemeral too.

Reviewer 4: ```suggestion
Work is ongoing to enable the Apple Silicon based runners as ephemeral runners to close this project from the macOS side. Once that is done the stretch goal for this project is to make our self-hosted Linux runners ephemeral too. Most of our Linux CI already uses ephemeral GitHub-hosted runners.
```

Trying to make it sound like it's coming in the near future without actually promising near future. I'm setting up the images again this week.

Also probably worth mentioning why Linux is less critical of an issue

---
title: Maintainer Projects
author: SMillerDev
---

Homebrew’s Project Leadership Committee has green-lit two paid projects by our maintainers this year and since both have hit some milestones recently we’d love to give you, our sponsors and users, an update on their progress.

## API Project

> Since version [3.6.0](./2022-09-07-homebrew-3.6.0) of Homebrew, we are now choosing to advertise the `HOMEBREW_INSTALL_FROM_API` environment variable to the general public. A lot of work was done by @Rylan12 to make the experience more stable and the API install method is bound to bring a big performance improvement to those who dare enable it.

Reviewer 6: I think?
```suggestion
Since version [3.6.0](/_posts/2022-09-07-homebrew-3.6.0.md) of Homebrew, we are now choosing to advertise the `HOMEBREW_INSTALL_FROM_API` environment variable to the general public. A lot of work was done by @Rylan12 to make the experience more stable and the API install method is bound to bring a big performance improvement to those who dare enable it.
```

Author: Well, it gets rendered to HTML so I'm not sure if this works.

Reviewer 3: ```suggestion
Since version [3.6.0]({% post_url 2022-09-07-homebrew-3.6.0 %}) of Homebrew, we are now choosing to advertise the `HOMEBREW_INSTALL_FROM_API` environment variable to the general public. A lot of work was done by @Rylan12 to make the experience more stable and the API install method is bound to bring a big performance improvement to those who dare enable it.
```

If you use this format it'll be checked by CI too

> Since version [3.6.0](./2022-09-07-homebrew-3.6.0.md) of Homebrew, we are now choosing to advertise the `HOMEBREW_INSTALL_FROM_API` environment variable to the general public. A lot of work was done by @Rylan12 to make the experience more stable and the API install method is bound to bring a big performance improvement to those who dare enable it.

Author: ```suggestion
Since version [3.6.0](./2022-09-07-homebrew-3.6.0) of Homebrew, we are now choosing to advertise the `HOMEBREW_INSTALL_FROM_API` environment variable to the general public. A lot of work was done by @Rylan12 to make the experience more stable and the API install method is bound to bring a big performance improvement to those who dare enable it.
```

> Since version [3.6.0]({% post_url 2022-09-07-homebrew-3.6.0 %}) of Homebrew, we are now choosing to advertise the `HOMEBREW_INSTALL_FROM_API` environment variable to the general public. A lot of work was done by @Rylan12 to make the experience more stable and the API install method is bound to bring a big performance improvement to those who dare enable it.

Reviewer 2: ```suggestion
Since version [3.6.0]({% post_url 2022-09-07-homebrew-3.6.0 %}) of Homebrew, we are now choosing to advertise the `HOMEBREW_INSTALL_FROM_API` environment variable to the general public. A lot of work was done by @Rylan12 to make the experience more stable and the API install method is bound to bring a big performance improvement to those who are willing to give it a try.
```

I don't think it matters much, but I worry that the word "dare" gives the impression that this is still in a very early beta with lots of bugs when in reality it's actually pretty close to being fully ready

> The goal of the `HOMEBREW_INSTALL_FROM_API` environment variable is to allow Homebrew users to install formulae and casks without needing to have Homebrew/homebrew-core and/or Homebrew/homebrew-cask tapped. This is intended to save space on the user’s machine and also to make commands like `brew update` much faster.

Author: ```suggestion
The goal of the `HOMEBREW_INSTALL_FROM_API` environment variable is to allow Homebrew users to install formulae and casks without needing to have Homebrew/homebrew-core and/or Homebrew/homebrew-cask tapped. This is intended to make `brew update` much faster and required less often and also to save space on the user’s machine .
```

> The goal of the `HOMEBREW_INSTALL_FROM_API` environment variable is to allow Homebrew users to install formulae and casks without needing to have Homebrew/homebrew-core and/or Homebrew/homebrew-cask tapped. This is intended to make `brew update` much faster and required less often and also to save space on the user’s machine .

Reviewer 2: ```suggestion
The goal of the `HOMEBREW_INSTALL_FROM_API` environment variable is to allow Homebrew users to install formulae and casks without needing to have Homebrew/homebrew-core and/or Homebrew/homebrew-cask tapped. This is intended to make `brew update` much faster and also to save space on the user’s machine.
```

I'm not sure the frequency is really affected, other than that running `brew update` is less painful now so it's not as big of a deal to run it

Author: @MikeMcQuaid this was your addition.

Reviewer 3: @Rylan12 @SMillerDev The frequency is not yet affected but we will definitely be able to significantly bump the default `HOMEBREW_AUTO_UPDATE_SECS` from 5 minutes once we're not relying on Homebrew/core or cask updates needing to come through as quickly.

> The goal of the `HOMEBREW_INSTALL_FROM_API` environment variable is to allow Homebrew users to install formulae and casks without needing to have Homebrew/homebrew-core and/or Homebrew/homebrew-cask tapped. This is intended to make `brew update` much faster and required less often and also to save space on the user’s machine .

Reviewer 2: ```suggestion
The goal of the `HOMEBREW_INSTALL_FROM_API` environment variable is to allow Homebrew users to install formulae and casks without needing to have Homebrew/homebrew-core and/or Homebrew/homebrew-cask tapped. This is intended to make `brew update` much faster and required less often and also to save space on the user’s machine.
```
